scripts/check_repo_root_size.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# If you need to add or remove from the repo root please change NUM_ITEMS_SHOULD_BE
+
+# Please take care to only add files or directories to the repo root unless they are
+# required to be in the repo root, otherwise please find a more appropriate location.
+
+NUM_ITEMS_SHOULD_BE=40
```

Comment:
Should we ignore directories and files that start with `.`?

Reply:
There are a lot of configuration files that do start with `.`

Reply:
How do we ignore files/directories that are captured in our `.gitignore`? What if this stage runs after something else and we have a `__pycache__` or `.ruff_cache`?

Reply:
That's a good point. We should exclude items in `.gitignore`.

Reply:
💯 I included all files because we do include configuration files starting with `.`. Since this is its own job, I believe it gets a fresh copy of the repo so I don't think we need to worry about something else running first, but please correct me if I'm wrong. See the Checkout great-expectations step in this job: https://dev.azure.com/great-expectations/great_expectations/_build/results?buildId=208968&view=logs&j=ab9c2685-3130-5a8e-cfa2-f6f8ba5ee4fb&t=a3589824-89d9-564f-3a8c-9b0da884204e

Reply:
Hoping to keep this a really simple check using a clean checkout of the repo, but if we need to worry about doing this check after more steps that sounds like a good approach 🙇